### PR TITLE
Add init multisignature parameters box

### PR DESCRIPTION
**Pull request by thdailong**
* Create a newpage for user to create a multisig transaction with input transaction and multisig parameters

FinalUI

### CODE REVIEW

**Comment by robert-zaremba** [comment]
Please add some margins. The images don't look beautiful, can you use antd icons? Please put more attention to the style  - it should look beautiful.

File: src/components/IconToolTip/IconWithToolTip.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<template>
+	<a-popover>
+		<template #content>
+			<p>
+				{{ data }}
+			</p>
+		</template>
+		<DeleteOutlined v-if="icon === 'Delete'" />
```
Comment (Megha-Dev-19) [inline src/components/IconToolTip/IconWithToolTip.vue]: I think we can save these values (like `Delete`) in constants.

File: src/constants/corlor.constant.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const INFO_ICON_COLOR = "#1890ff";
+export const CHECK_ICON_COLOR = "#38cb15";
+export const CLOSE_ICON_COLOR = "#f70300";
```
Comment (Megha-Dev-19) [inline src/constants/corlor.constant.ts]: file name spelling is incorrect, it should be `color.constant.ts`

File: src/constants/success.constant.ts [hunk 3]
```diff
@@ -20,5 +20,9 @@ export const successMessage = (key: string) => {
 	message.loading({ content: "Loaded", key: key });
 };
 
+export const successCopyAddress = (address: string) => {
+	message.success("Address [" + address + "] copied!");
+};
+
```
Comment (Megha-Dev-19) [inline src/constants/success.constant.ts:26]: let's create a new function for success messages and in that we should have one for copy, we can even use the `openSuccessNotificationWithIcon` for this.

File: src/pages/InitMsig.vue [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+import { WebMode, WallectConnectSession } from "@algo-builder/web";
+import { JsonPayload } from "@algo-builder/web/build/algo-signer-types";
+import algosdk from "algosdk";
+import WalletStore from "@/store/WalletStore";
+import IconWithToolTip from "@/components/IconToolTip/IconWithToolTip.vue";
+
+let id = 0;
+
+export default defineComponent({
+	name: "Init multisignature transaction UI",
```
Comment (Megha-Dev-19) [inline src/pages/InitMsig.vue]: name shouldn't be this long, it should be small and understandable, something like `Msig Tx Creator`

File: src/pages/InitMsig.vue [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+			JSON: "JSON",
+			MSG_PACK: "MSG_PACK(base64)",
+		};
+
+		const walletStore = WalletStore();
+
+		const key = ref("tab1");
+
+		const onTabChange = (value: string, type: string) => {
+			console.log(value, type);
```
Comment (Megha-Dev-19) [inline src/pages/InitMsig.vue]: let's remove the log

File: src/pages/InitMsig.vue [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+						break;
+					}
+					case WalletType.WALLET_CONNECT: {
+						let signWalletConnect = this.walletStore
+							.webMode as WallectConnectSession;
+
+						let trxs = algosdk.decodeUnsignedTransaction(
+							convertBase64ToUnit8Array(txnBase64)
+						);
+						let signedJson = await signWalletConnect.signTransactionGroup([
```
Comment (Megha-Dev-19) [inline src/pages/InitMsig.vue]: after signing we should show a success message

File: src/pages/MsigUI.vue [hunk 7]
```diff
@@ -105,85 +109,107 @@ export default defineComponent({
...
-				default: {
-					console.log("Invalid wallet type connected");
-					break;
+			try {
+				switch (this.walletStore.walletKind) {
+					case WalletType.MY_ALGO: {
+						let signMyAlgo = this.walletStore.webMode as MyAlgoWalletSession;
+
+						let trxs = algosdk.decodeUnsignedTransaction(
+							convertBase64ToUnit8Array(txnBase64)
```
Comment (Megha-Dev-19) [inline src/pages/MsigUI.vue]: I believe this is same as `sign` method in InitMsig, so let's add a common function and use it in both places.

**Review by Megha-Dev-19 — CHANGES_REQUESTED** [review]
Reviewed.

File: src/pages/MsigUI.vue (hunk 7 above)
Comment (thdailong) [reply src/pages/MsigUI.vue]: Well this one for `myalgo` and `wallet_connect`, I have not finished the feature signing multisig transaction due to error and not support signing! This one is for testing signing a single signature transaction

File: package.json [hunk 8]
```diff
@@ -19,8 +19,8 @@
 		"test": "echo 'no test'"
 	},
 	"dependencies": {
-		"@algo-builder/algob": "^4.0.0",
-		"@algo-builder/web": "^4.0.0",
+		"@algo-builder/algob": "^5.0.0",
+		"@algo-builder/web": "^5.0.0",
```
Comment (robert-zaremba) [inline package.json]: let's use `5.0.1`

**Review by robert-zaremba — APPROVED** [review]

File: src/pages/MsigUI.vue (hunk 7 above)
Comment (Megha-Dev-19) [reply src/pages/MsigUI.vue]: okay

File: src/constants/constant.ts [hunk 9]
```diff
@@ -19,3 +19,10 @@ export const tabList = [
 		tab: formatTypes.MSG_PACK,
 	},
 ];
+
+export const action = {
+	delete: "Delete",
+	edit: "Edit",
+	save: "Save",
+	copy: "Copy",
+};
```
Comment (Megha-Dev-19) [inline src/constants/constant.ts]:
```suggestion
export const Action = {
	DELETE: "Delete",
	EDIT: "Edit",
	SAVE: "Save",
	COPY: "Copy",
};
```

File: src/constants/error.constant.ts [hunk 10]
```diff
@@ -25,3 +25,7 @@ export const UNSUCCESSFUL = "Unsuccessful";
 export const wrongAddress = "Wrong multisigature parameters!";
 export const wrongAddressDes =
 	"Address generate from multisignature parameters is not equal to the sender of the transaction";
+
+export const noWallet = "You need to login your wallet to sign transaction!";
+export const notSupportWallet =
+	"We are currently unable to support MyAlgo and Wallet Connect to sign multisigature transaction";
```
Comment (Megha-Dev-19) [inline src/constants/error.constant.ts]: Since these are global constants it should start with capital letters looking at the convention used across files

**Review by Megha-Dev-19 — APPROVED** [review]
Left few suggestions. Pre approving.

**Pull request merged**